### PR TITLE
fix(inx): node config updates

### LINKED ISSUE #1108: Write past node config into MongoDB
Now that we listen to confirmed milestones, we should also push the historical protocol parameters to the database. Right now we only push protocol parameters that are equal to or newer than the starting of the sync process.

### PR DESCRIPTION
**Pull request by Alexandcoats**
## Linked Issues

* Closes #1108 

## Notes to Reviewer

As a reviewer, please pay particular attention to the following areas when reviewing this PR and tick the above boxes after you have completed the steps.

#### INX Changes
* [ ] Run chronicle using an INX connection.
* [x] Ensure that node configs older than the `starting_index` are still written (Alex6323: I think that's covered by the tests).

#### Database Changes
* [x] Review database queries for correctness/conciseness.
* [x] Ensure queries are supported by indexes if needed.
* [x] Check for breaking changes in the data model and matching (conventional) commit message prefix.

### CODE REVIEW

File: tests/node_configuration.rs [hunk 1]
```diff
@@ -93,19 +93,6 @@ mod test_rand {
...
-            .await
-            .unwrap();
-        assert_eq!(node_configuration.count().await.unwrap(), 1);
-        assert_eq!(&latest_config.config.base_token.name, "Shimmer");
-        assert_eq!(&latest_config.config.base_token.ticker_symbol, "SMR");
-        assert_eq!(&latest_config.config.base_token.unit, "SMR");
-        assert_eq!(&latest_config.config.base_token.subunit, "glow");
-        assert_eq!(latest_config.config.base_token.decimals, 6);
-        assert!(latest_config.config.base_token.use_metric_prefix);
-
```
Comment (grtlr) [inline tests/node_configuration.rs:108]: I don't think we should delete this test case. We should rather "sandwich" a meaningless configuration _t1_ (where nothing has changed) between two configurations at _t0_ and _t2_. Ideally, _t0_ would be inserted after _t2_ to mimic our logic for catching up.

File: src/db/collections/protocol_update.rs [hunk 2]
```diff
@@ -92,17 +92,17 @@ impl ProtocolUpdateCollection {
     }
 
     /// Add the protocol parameters to the list if the protocol parameters have changed.
-    pub async fn update_latest_protocol_parameters(
+    pub async fn upsert_protocol_parameters(
```
Comment (Alex6323) [inline src/db/collections/protocol_update.rs:95]: Should `insert_protocol_parameters` become private now as both cases (insertion, updating) are handled by this method, and the user can't keep track of whether it was inserted already anyway?

File: src/db/collections/configuration_update.rs [hunk 3]
```diff
@@ -64,40 +68,18 @@ impl ConfigurationUpdateCollection {
...
-            if latest_config.ledger_index >= ledger_index {
-                return Ok(());
-            } else if latest_config.config != config {
-                self.insert_one(ConfigurationUpdateDocument { ledger_index, config }, None)
-                    .await?;
-            }
-        } else {
-            self.insert_one(ConfigurationUpdateDocument { ledger_index, config }, None)
-                .await?;
+        let params = self.get_node_configuration_for_ledger_index(ledger_index).await?;
```
Comment (Alex6323) [inline src/db/collections/configuration_update.rs:71]: `params` ~> `node_config`
`latest_config` ~> `node_config`

File: src/db/collections/configuration_update.rs [hunk 4]
```diff
@@ -64,40 +68,18 @@ impl ConfigurationUpdateCollection {
...
+        if params.is_none()
+            || params
+                .map(|latest_config| latest_config.config != config)
+                .unwrap_or_default()
```
Comment (Alex6323) [inline src/db/collections/configuration_update.rs:75]: `if !matches!(node_config, Some(node_config) if node_config == config) { .. }`

File: src/db/collections/configuration_update.rs [hunk 5]
```diff
@@ -64,40 +68,18 @@ impl ConfigurationUpdateCollection {
...
+        {
+            self.update_one(
+                doc! { "_id": ledger_index },
+                doc! { "$set": mongodb::bson::to_bson(&config).unwrap() },
```
Comment (Alex6323) [inline src/db/collections/configuration_update.rs:79]: Undocumented `unwrap`.

File: src/db/collections/protocol_update.rs [hunk 6]
```diff
@@ -73,36 +77,26 @@ impl ProtocolUpdateCollection {
...
+                .unwrap_or_default()
+        {
+            self.update_one(
+                doc! { "_id": ledger_index },
+                doc! { "$set": {
+                    "parameters": mongodb::bson::to_bson(&parameters).unwrap()
+                } },
+                UpdateOptions::builder().upsert(true).build(),
+            )
+            .await?;
```
Comment (Alex6323) [inline src/db/collections/protocol_update.rs:99]: here as well ... I think `!matches!` is still preferable ... and the undocumented `unwrap`

File: src/db/collections/configuration_update.rs (hunk 5 above)
Comment (Alexandcoats) [reply src/db/collections/configuration_update.rs:79]: This isn't exactly the first one for `to_bson`.

**Review by Alex6323 — APPROVED** [review]

**Pull request merged**